test/functional/date/parse-date.js

```diff
@@ -139,6 +169,27 @@ QUnit.test( "should parse date presets", function( assert ) {
 	assertParseDate( assert, "9/15/10", { date: "short" }, date );
 });
 
+QUnit.test( "should parse date correctly in leap year", function( assert ) {
+	var OrigDate;
+
+	extraSetup();
+
+	/* globals Date:true */
+	// Use a leap year and a day of month greater than 28
+	OrigDate = Date;
+	Date = FakeDate;
+	FakeDate.today = new Date( 2016, 0, 29 );
+
+	date = new Date( 2015, 1, 15 );
+	date = startOf( date, "day" );
+	assertParseDate( assert, "Wednesday, February 15, 2015", { date: "full" }, date );
+	assertParseDate( assert, "February 15, 2015", { date: "long" }, date );
+	assertParseDate( assert, "Feb 15, 2015", { date: "medium" }, date );
+	assertParseDate( assert, "2/15/15", { date: "short" }, date );
+
+	Date = OrigDate;
+});
+
```

Comment:
Please, could you move this test from `functional` to `unit` ([file](https://github.com/jquery/globalize/blob/master/test/unit/date/parse.js#L148))? Unit tests are better to keep the specifics while functional tests are better to test bigger integrations.